Python/_warnings.c

```diff
@@ -883,6 +883,13 @@ get_source_line(PyObject *module_globals, int lineno)
     PyObject *source_list;
     PyObject *source_line;
 
+    if (!PyDict_Check(module_globals)) {
```

Comment:
I would move this check into `warnings_warn_explicit()`.

Reply:
fixed